### PR TITLE
Feature 1150/send email notifications to guild leaders when changes are made to membership

### PR DESCRIPTION
**Pull request by HootDunk**
This will send emails to guild leads when changes are made to guild membership that they did not make themselves.  Tests that mock the sendEmail function are included.

### CODE REVIEW

**Review by moshirim — APPROVED** [review]

File: server/src/main/java/com/objectcomputing/checkins/notifications/email/EmailSender.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 package com.objectcomputing.checkins.notifications.email;
 
 public interface EmailSender {
-    public void sendEmail(String subject, String content);
+    public void sendEmail(String subject, String content, String... recipients);
```
Comment (jahanner) [inline server/src/main/java/com/objectcomputing/checkins/notifications/email/EmailSender.java:4]: should this be String ...recipients or am I reading that wrong?

Comment (HootDunk) [reply server/src/main/java/com/objectcomputing/checkins/notifications/email/EmailSender.java:4]: The three dots in java are called Varargs and they work a bit different then the spread operator for JavaScript.  I saw it being used this way in one of the commits in the feedback branch and had to look it up myself (it seemed like a good use though so I repeated it).  Java treats the recipients variable as an array of strings but when you call the method, if you only have one recipient, you can pass the string directly and skip putting it into an array. You can also explicitly pass multiple String recipients and Java will convert it to an array for you. Of course you can also just pass in an array as well if that's what you're working with already. 

So all of the following are valid ways to call the method:
sendEmail (subject, content, "someEmail@gmail.com");
sendEmail (subject, content, "someEmail@gmail.com", "anotherEmail@gmail.com");

String[] emails = new String[] {"someEmail@gmail.com", "anotherEmail@gmail.com", "aThirdEmail@gmail.com"};
sendEmail(subject, content, emails);

Comment (jahanner) [reply server/src/main/java/com/objectcomputing/checkins/notifications/email/EmailSender.java:4]: Nice, I'll have to keep Varargs in mind then

File: server/src/test/java/com/objectcomputing/checkins/services/guild/GuildControllerTest.java [hunk 2]
```diff
@@ -17,14 +18,18 @@
 import io.micronaut.http.client.HttpClient;
 import io.micronaut.http.client.annotation.Client;
 import io.micronaut.http.client.exceptions.HttpClientResponseException;
+import jnr.ffi.annotations.In;
```
Comment (mkimberlin) [inline server/src/test/java/com/objectcomputing/checkins/services/guild/GuildControllerTest.java:21]: I'm guessing this was an accident?

**Review by mkimberlin — APPROVED** [review]

File: server/src/test/java/com/objectcomputing/checkins/services/guild/GuildControllerTest.java (hunk 2 above)
Comment (HootDunk) [reply server/src/test/java/com/objectcomputing/checkins/services/guild/GuildControllerTest.java:21]: Yep, will remove.

**Pull request merged**